mmdet/core/general_data/general_data.py

```diff
@@ -0,0 +1,497 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+import itertools
+
+import numpy as np
+import torch
+
+from mmdet.utils.util_mixins import NiceRepr
+
+
+class GeneralData(NiceRepr):
+    """A general data structure of OpenMMlab.
+
+    A data structure that stores the meta information,
+    the annotations of the images or the model predictions,
+    which can be used in communication between components.
+
+    The attributes in `GeneralData` are divided into two parts,
+    the `meta_info_fields` and the `data_fields` respectively.
+
+        - `meta_info_fields`: Usually contains the
+          information about the image such as filename,
+          image_shape, padding_shape, etc. All attributes in
+          it are immutable once set,
+          but the user can add new meta information with
+          `set_meta` function, all information can be accessed
+          with methods `meta_info_keys`, `meta_info_values`,
+          `meta_info_items`.
+
+        - `data_fields`: Annotations or model predictions are
+          stored. The attributes can be accessed or modified by
+          dict-like or object-like operations, such as
+          `.` , `[]`, `in`, `del`, `pop(str)` `get(str)`, `keys()`,
+          `values()`, `items()`. Users can also apply tensor-like methods
+          to all obj:`torch.Tensor` in the `data_fileds`,
+          such as `.cuda()`, `.cpu()`, `.numpy()`, `device`, `.to()`
+          `.detach()`, `.numpy()`
+
+    Args:
+        meta_info (dict): A dict contains the meta information
+            of single image. such as `img_shape`, `scale_factor`, etc.
+            Default: None.
+        data (dict): A dict contains annotations of single image or
+            model predictions. Default: None.
+
+    Examples:
+        >>> from mmdet.core import GeneralData
+        >>> img_meta = dict(img_shape=(800, 1196, 3), pad_shape=(800, 1216, 3))
+        >>> results = GeneralData(meta_info=img_meta)
+        >>> img_shape in results
+        True
+        >>> results.det_labels = torch.LongTensor([0, 1, 2, 3])
+        >>> results["det_scores"] = torch.Tensor([0.01, 0.1, 0.2, 0.3])
+        >>> print(results)
```

Comment:
Is it necessary to print the data type at the same time？

Reply:
This is the design of `NiceRepr`